libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs

```diff
@@ -104,12 +104,12 @@ private static JToken GetEntityValue(EntityRecommendation entity)
 
             if (entity.Type.StartsWith("builtin.datetimeV2."))
             {
-                return new JValue(entity.Resolution?.Values != null && entity.Resolution.Values.Count > 0
-                            ? ((IDictionary<string, object>)((IList<object>)entity.Resolution.Values.First()).First())["timex"]
-                            : entity.Resolution);
+                return entity.Resolution?.Values != null && entity.Resolution.Values.Count > 0
```

Comment:
This code seems very complex. Could you turn it from a single return statement into a nice set of clear code that's more-or-less self documenting?

Reply:
Sure